django/core/management/__init__.py

```diff
@@ -138,6 +138,7 @@ def get_actions(parser):
     # Any required arguments which are passed in via **options must be passed
     # to parse_args().
     parse_args += [
+        min(opt.option_strings) if isinstance(opt, _StoreConstAction) else
```

Comment:
As far as I'm aware we should pass a constant option only if `arg_options[opt.dest]` is `True`

Reply:
Do you mean `if opt.dest in arg_options`? 
So, by considering `append_const` and `count`, would be like:

```python
    parse_args += [
        min(opt.option_strings) if (
            isinstance(opt, (_AppendConstAction, _CountAction, _StoreConstAction)) and
            opt.dest in arg_options
        ) else
        '{}={}'.format(min(opt.option_strings), arg_options[opt.dest])
        for opt in parser_actions if (
            opt.dest in options and
            (opt.required or opt in mutually_exclusive_required_options)
        )
    ]
```

Reply:
No, `arg_options[opt.dest]` should be set to `True`, passing `flag=False` means that we don't want to include this option. That's how it works for arguments without groups.

Reply:
Sorry I edited your comment by mistake.

Reply:
> Sorry I edited your comment by mistake.

No worries.

I cannot understand what is your mean TBH. if we have a management command with arguments without groups:
```python
class Command(BaseCommand):
    def add_arguments(self, parser):
        parser.add_argument('--flag1', action='store_true')                               

    def handle(self, *args, **options):
        print('args: ', args)
        print('opitons: ', options)
```
Here is the output in different cases:
`./manage.py test`

```
args:  ()
opitons:  {..., 'flag1': False}
```

`./manage.py test --flag1`

```
args:  ()
opitons:  {..., 'flag1': True}
```

`call_command('test')`

```
args:  ()
opitons:  {..., 'flag1': False}
```

`call_command('test', '--flag1')`

```
args:  ()
opitons:  {..., 'flag1': True}
```

`call_command('test', flag1=True)`

```
args:  ()
opitons:  {..., 'flag1': True}
```

`call_command('test', flag1=False)`

```
args:  ()
opitons:  {..., 'flag1': False}
```

`call_command('test', flag1=0)`

```
args:  ()
opitons:  {..., 'flag1': 0}
```

`call_command('test', flag1='test')`

```
args:  ()
opitons:  {..., 'flag1': 'test'}
```

and I think with the new changes this will happen for arguments with groups.

Reply:
It's not about `options` but about argument that we pass to `parser.parse_args()`, you cannot pass `--flag1` when user call `call_command('test', flag1=False)`. Currently truthy values as treated as "flag selected" and we should keep this behavior for mutually exclusive groups, so:
- `call_command('mutually_exclusive_required_with_boolean', flag_true=True)` -> `--flag_true` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_true='x')` -> `--flag_true` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_true=78)` -> `--flag_true` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_true=0)` -> `''` :red_circle:
- `call_command('mutually_exclusive_required_with_boolean', flag_true=False)` -> `''` :red_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_true=[])` -> `''` :red_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_false=True)` -> `--flag_false` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_false='x')` -> `--flag_false` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_false=78)` -> `--flag_false` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_false=0)` -> `''` :red_circle:
- `call_command('mutually_exclusive_required_with_boolean', flag_false=False)` -> `''` :red_circle: 
- `call_command('mutually_exclusive_required_with_boolean', flag_false=[])` -> `''` :red_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const=True)` -> `--const` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const='x')` -> `--const` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const=78)` -> `--const` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const=41)` -> `--const` :green_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const=0)` -> `''` :red_circle:
- `call_command('mutually_exclusive_required_with_boolean', const=False)` -> `''` :red_circle: 
- `call_command('mutually_exclusive_required_with_boolean', const=[])` -> `''` :red_circle:

Reply:
@hramezani I checked this again and it looks that I was wrong, we always pass value directly to the command, even if an invalid value is passed. Sorry for misleading. I've reverted this change.